src/Swift.Bindings/src/Emitter/StringEmitter/Handler/PropertyHandler.cs

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.CodeDom.Compiler;
+
+namespace BindingsGeneration;
+
+/// <summary>
+/// Factory class for creating instances of PropertyHandler.
+/// </summary>
+public class PropertyHandlerFactory : IFactory<BaseDecl, IPropertyHandler>
+{
+    public bool Handles(BaseDecl decl)
+    {
+        return decl is PropertyDecl propertyDecl && propertyDecl.Accessors.Any();
+    }
+
+    public IPropertyHandler Construct()
+    {
+        return new PropertyHandler();
+    }
+}
+
+/// <summary>
+/// Handler class for property declarations that generates the binding code for Swift properties.
+/// </summary> 
+public class PropertyHandler : BaseHandler, IPropertyHandler
+{
+    private readonly MethodHandler _methodHandler = new();
+
+    /// <inheritdoc/>
+    public IEnvironment Marshal(BaseDecl baseDecl, ITypeDatabase typeDatabase)
+    {
+        if (baseDecl is not PropertyDecl propertyDecl)
+        {
+            throw new ArgumentException("The provided decl must be a PropertyDecl.", nameof(baseDecl));
+        }
+        return new PropertyEnvironment(propertyDecl, typeDatabase);
+    }
+
+    /// <inheritdoc/>
+    public void Emit(CSharpWriter csWriter, SwiftWriter swiftWriter, IEnvironment env, Conductor conductor)
+    {
+        var propertyEnv = (PropertyEnvironment)env;
+        var propertyDecl = propertyEnv.PropertyDecl;
+
+        var typeRecord = propertyEnv.TypeDatabase.GetTypeRecordOrThrow(propertyDecl.SwiftTypeSpec);
+
+        var staticModifier = propertyDecl.IsStatic ? "static " : "";
```

Comment:
Do we support static properties? I think it is not part of the [proposal](https://github.com/dotnet/runtimelab/pull/2988) and I'm unsure what quirks there are. I think we should assert if we encounter something we don't support (or partially) yet.